src/server/db_slice.cc

```diff
@@ -847,21 +859,19 @@ OpResult<int64_t> DbSlice::UpdateExpire(const Context& cntx, PrimeIterator prime
   }
 }
 
-std::pair<PrimeIterator, bool> DbSlice::AddOrUpdateInternal(const Context& cntx,
-                                                            std::string_view key, PrimeValue obj,
-                                                            uint64_t expire_at_ms,
-                                                            bool force_update) noexcept(false) {
+DbSlice::AddOrFindResult DbSlice::AddOrUpdateInternal(const Context& cntx, std::string_view key,
```

Comment:
you can now updating the expire iterator returned from this function

Reply:
Sorry, I'm not sure what you mean? Did you find a regression in my PR?

Reply:
AddOrFindResult contains expire iterator. 
below you have a code that can inserts to db.expire but you do not update the expire iterator returned from this funciton.
Also in the flow below if we need add expire info you can check if the expire_it from AddOrFind is valid and if so update it otherwise do the insert, this way you will not do find twice on expire db

Reply:
oh wow, nice! great comment.